ml-agents/mlagents/trainers/settings.py

```diff
@@ -151,6 +151,92 @@ class CuriositySettings(RewardSignalSettings):
     learning_rate: float = 3e-4
 
 
+class ParameterRandomizationType(Enum):
+    UNIFORM: str = "uniform"
+    GAUSSIAN: str = "gaussian"
+    MULTIRANGEUNIFORM: str = "multirangeuniform"
+
+    def to_settings(self) -> type:
+        _mapping = {
+            ParameterRandomizationType.UNIFORM: UniformSettings,
+            ParameterRandomizationType.GAUSSIAN: GaussianSettings,
+            ParameterRandomizationType.MULTIRANGEUNIFORM: MultiRangeUniformSettings,
+        }
+        return _mapping[self]
+
+    def to_float(self) -> float:
+        _mapping = {
+            ParameterRandomizationType.UNIFORM: 0.0,
+            ParameterRandomizationType.GAUSSIAN: 1.0,
+            ParameterRandomizationType.MULTIRANGEUNIFORM: 2.0,
+        }
+        return _mapping[self]
+
+
+@attr.s(auto_attribs=True)
+class ParameterRandomizationSettings:
+    @staticmethod
+    def structure(d: Mapping, t: type) -> Any:
+        """
+        Helper method to structure a Dict of ParameterRandomizationSettings class. Meant to be registered with
+        cattr.register_structure_hook() and called with cattr.structure(). This is needed to handle
+        the special Enum selection of ParameterRandomizationSettings classes.
+        """
+        if not isinstance(d, Mapping):
+            raise TrainerConfigError(
+                f"Unsupported parameter randomization configuration {d}."
+            )
+        d_final: Dict[str, List[float]] = {}
+        for param, param_config in d.items():
+            for key, val in param_config.items():
+                enum_key = ParameterRandomizationType(key)
+                t = enum_key.to_settings()
+                d_final[param] = strict_to_cls(val, t).to_float()
+        return d_final
+
+
+@attr.s(auto_attribs=True)
+class UniformSettings(ParameterRandomizationSettings):
+    min_value: float = 1.0
+    max_value: float = 1.0
+
+    def to_float(self) -> List[float]:
+        if self.min_value > self.max_value:
+            raise TrainerConfigError(
+                "Minimum value is greater than maximum value in uniform sampler."
+            )
+        return [0.0, self.min_value, self.max_value]
+
+
+@attr.s(auto_attribs=True)
+class GaussianSettings(ParameterRandomizationSettings):
+    mean: float = 1.0
+    st_dev: float = 1.0
+
+    def to_float(self) -> List[float]:
+        return [1.0, self.mean, self.st_dev]
+
+
+@attr.s(auto_attribs=True)
+class MultiRangeUniformSettings(ParameterRandomizationSettings):
+    intervals: List[List[float]] = [[1.0, 1.0]]
+
+    def to_float(self) -> List[float]:
+        floats: List[float] = []
+        for interval in self.intervals:
+            if len(interval) != 2:
```

Comment:
These should go in the validators for `interval` (there's one in `TrainerSettings` for the batch size / sequence length) so that the parameters are validated even when we don't call `to_float`.

Reply:
Fixed.